src/Workspaces/Core/Portable/Workspace/Host/Mef/MefHostServices.cs

```diff
@@ -67,6 +79,12 @@ private class WithMetadataImporter<TExtension, TMetadata>
 
         #region Defaults
 
+        internal static void HookServiceCreation(CreationHook hook)
+        {
+            s_CreationHook = hook;
+            s_defaultHost = null;
```

Comment:
Comment why. I can guess, but I'm not necessarily the next person to touch this. :smile:

Reply:
➡️ Added comments in 37a1d2c